### PR TITLE
Add 1.17 deploy feature to skuba

### PR DESCRIPTION
**Pull request by evrardjp**
This adds:
- 1.17 in the available versions for skuba, pointing to the right containers
- Test the 1.16->1.17 in CI

### CODE REVIEW

File: internal/pkg/skuba/kubernetes/versions.go [hunk 1]
```diff
@@ -79,6 +79,27 @@ type KubernetesVersions map[string]KubernetesVersion
...
 var (
 	supportedVersions = KubernetesVersions{
+                "1.17.4": KubernetesVersion{
+			ComponentHostVersion: ComponentHostVersion{
+				KubeletVersion:          "1.17.4",
+				ContainerRuntimeVersion: "1.17.0",
+			},
+			ComponentContainerVersion: ComponentContainerVersion{
+				Hyperkube: &ContainerImageTag{Name: "hyperkube", Tag: "v1.17.4"},
+				Etcd:      &ContainerImageTag{Name: "etcd", Tag: "3.3.15"},
```
Comment (ereslibre) [inline internal/pkg/skuba/kubernetes/versions.go]:
```suggestion
				Etcd:      &ContainerImageTag{Name: "etcd", Tag: "3.4.3"},
```

File: internal/pkg/skuba/kubernetes/versions.go [hunk 2]
```diff
@@ -79,6 +79,27 @@ type KubernetesVersions map[string]KubernetesVersion
...
+				CoreDNS:   &ContainerImageTag{Name: "coredns", Tag: "1.6.2"},
```
Comment (ereslibre) [inline internal/pkg/skuba/kubernetes/versions.go]:
```suggestion
				CoreDNS:   &ContainerImageTag{Name: "coredns", Tag: "1.6.5"},
```

**Comment by evrardjp** [comment]
To clarify for the future, the reason why the bump of coredns and etcd, it's to match the upstream recommended versions.

**Comment by evrardjp** [comment]
I will work with @mssola on updating those images, when they are available, I will update this.

**Comment by evrardjp** [comment]
Updated. We'll discuss the pattern package tomorrow.

**Comment by evrardjp** [comment]
This has removed the cilium requirements from the equation, dealing with them in the other PR.

**Pull request merged**